src/css/wizard-navigation-bar.scss

```diff
@@ -140,120 +120,155 @@ $wz-param-indicator-color: null !global;
       @content;
     }
 
     &.navigable a:hover .step-indicator {
       $wz-param-indicator-state: 'hover' !global;
-      $wz-param-indicator-color: $wz-color-default !global;
-      @include wz-state-circle($width, $height, $border-width);
+      $wz-param-step-state: 'default' !global;
+      @include wz-circle($width, $height, $border-width);
       @content;
     }
     &.navigable.optional a:hover .step-indicator {
       $wz-param-indicator-state: 'hover' !global;
-      $wz-param-indicator-color: $wz-color-optional !global;
+      $wz-param-step-state: 'optional' !global;
       @content;
     }
     &.navigable.done a:hover .step-indicator {
       $wz-param-indicator-state: 'hover' !global;
-      $wz-param-indicator-color: $wz-color-done !global;
+      $wz-param-step-state: 'done' !global;
       @content;
     }
     &.navigable.current a:hover .step-indicator {
       $wz-param-indicator-state: 'hover' !global;
-      $wz-param-indicator-color: $wz-color-current !global;
+      $wz-param-step-state: 'current' !global;
       @content;
     }
     &.navigable.editing a:hover .step-indicator {
       $wz-param-indicator-state: 'hover' !global;
-      $wz-param-indicator-color: $wz-color-editing !global;
+      $wz-param-step-state: 'editing' !global;
       @content;
     }
     // 'completed' class takes priority
     // https://github.com/madoar/angular-archwizard/pull/221
     &.navigable.completed a:hover .step-indicator {
       $wz-param-indicator-state: 'hover' !global;
-      $wz-param-indicator-color: $wz-color-done !global;
+      $wz-param-step-state: 'done' !global;
       @content;
     }
   }
 }
 
 
-// Helper mixin to define indicator styles like 'small', 'large-filled-symbols' etc. for the given layout
-//
-// Arguments:
-//   $layout - 'horizontal' | 'vertical'
-//
-@mixin wz-define-styles($layout) {
-  &.small ul.steps-indicator {
-    @include wz-define-style($layout, $width: $wz-small-dot-width, $height: $wz-small-dot-height, $border-width: 0) {
-      @include wz-state-circle-with-background($wz-param-indicator-color);
-    }
-  }
-
-  &.large-filled ul.steps-indicator {
-    @include wz-define-style($layout, $border-width: 0) {
-      @include wz-state-circle-with-background($wz-param-indicator-color);
-    }
-  }
-
-  &.large-empty ul.steps-indicator {
-    @include wz-define-style($layout) {
-      @include wz-state-circle-with-border($wz-param-indicator-color);
-    }
-  }
-
-  &.large-filled-symbols ul.steps-indicator {
-    @include wz-define-style($layout, $border-width: 0) {
-      @include wz-state-circle-with-background-and-content($wz-param-indicator-color);
-    }
-  }
-
-  &.large-empty-symbols ul.steps-indicator {
-    @include wz-define-style($layout) {
-      @include wz-state-circle-with-border-and-content($wz-param-indicator-color);
+// Define wizard styles specific to the configurated layouts (`$wz-layouts`) and indicator styles (`$wz-indicator-styles`)
+@mixin wz-define-styles() {
```

Comment:
This mixin is kind of like black magic to me. I'm not that experienced with css and even less with scss, can you add some more content to the description/comment of the mixin to make it more clear what it does

Reply:
Added comments in d2274c9